play/tests/unittests/PhaseResults.test.tsx

```diff
@@ -0,0 +1,165 @@
+    // tests that new score has a value of 0
+    expect(apiClient.updateTeam).toHaveBeenCalledWith({
+      id: gameSession.teams![0].id,
+      score: 120,
+    });
+  });
+
+   // tests if player has answered correctly on phase 1 (starting score: 120, ending score: 130)
+   it('Phase 1, correct answer', async () => {
+    // mock gameSession with team that answered incorrectly on first question
+    const gameSession = GameSessionParser.gameSessionFromAWSGameSession(
+      mockPhaseOneTenPointsGameSession as IAWSGameSession
+    ) as IGameSession;
+    const mockCurrentQuestion = gameSession.questions[gameSession.currentQuestionIndex!];
+    const mockAnswerChoices = getAnswerChoices(mockCurrentQuestion);
+    act(() => {
+      renderWithThemeRouterTranslation(gameSession, mockAnswerChoices);
+    });
+
+    await waitFor(() => expect(apiClient.updateTeam).toHaveBeenCalled());
+
+    // tests that new score indicator has value of +10
+    expect(apiClient.updateTeam).toHaveBeenCalledWith({
+      id: gameSession.teams![0].id,
+      score: 130,
+    });
+  });
+
+    // tests if player has answered correctly on phase 1 (starting score: 120, ending score: 130)
+    it('Phase 2, unpopular trick answer', async () => {
+      // mock gameSession with team that answered incorrectly on first question
+      const gameSession = GameSessionParser.gameSessionFromAWSGameSession(
+        mockPhaseTwoUnpopularAnswerGamesession as IAWSGameSession
+      ) as IGameSession;
+      const mockCurrentQuestion = gameSession.questions[gameSession.currentQuestionIndex!];
+      const mockAnswerChoices = getAnswerChoices(mockCurrentQuestion);
+      act(() => {
+        renderWithThemeRouterTranslation(gameSession, mockAnswerChoices);
+      });
+  
+      await waitFor(() => expect(apiClient.updateTeam).toHaveBeenCalled());
+  
+      // tests that new score indicator has value of +10
+      expect(apiClient.updateTeam).toHaveBeenCalledWith({
+        id: gameSession.teams![0].id,
+        score: 120,
+      });
+    });
+
+     // tests if player has answered correctly on phase 1 (starting score: 120, ending score: 130)
+     it('Phase 2, popular trick answer', async () => {
+      // mock gameSession with team that answered incorrectly on first question
+      const gameSession = GameSessionParser.gameSessionFromAWSGameSession(
+        mockPhaseTwoPopularAnswerGamesession as IAWSGameSession
+      ) as IGameSession;
+      const mockCurrentQuestion = gameSession.questions[gameSession.currentQuestionIndex!];
+      const mockAnswerChoices = getAnswerChoices(mockCurrentQuestion);
+      act(() => {
+        renderWithThemeRouterTranslation(gameSession, mockAnswerChoices);
+      });
+  
+      await waitFor(() => {
+        expect(apiClient.updateTeam).toHaveBeenCalled()});
+  
+      // tests that new score indicator has value of +10
+      expect(apiClient.updateTeam).toHaveBeenCalledWith({
+        id: gameSession.teams![0].id,
```

Comment:
Same comment, in general, in tests, we should always confirm `teams` is not `null` and then use forced unwrap `!`

Reply:
corrected